src/components/lyne-tab-group/lyne-tab-group.tsx

```diff
@@ -0,0 +1,284 @@
+   * the tab group, the first enabled tab will be selected.
+   */
+  @Prop({
+    mutable: true
+  }) public selectedIndex = 0;
+
+  /**
+   * Disable tab by index
+   */
+  @Method()
+  public async disableTab(tabIndex: number): Promise<void> {
+    await this.tabs[tabIndex].tabGroupActions.disable();
+  }
+
+  /**
+   * Enable tab by index
+   */
+  @Method()
+  public async enableTab(tabIndex: number): Promise<void> {
+    await this.tabs[tabIndex].tabGroupActions.enable();
+  }
+
+  /**
+   * Activate tab by index
+   */
+  @Method()
+  public async activateTab(tabIndex: number): Promise<void> {
+    await this.tabs[tabIndex].tabGroupActions.toggle();
+  }
+
+  @Element() private _element: HTMLElement;
+
+  public tabs: InterfaceLyneTabGroupTab[];
+  private _lastUId = 0;
+  private _observer = new MutationObserver(this._onTabAttributesChange.bind(this));
+
+  public render(): JSX.Element {
+    return (
+      <Host>
+        <div class='tab-group' role='tablist'>
+          <slot name='tab-bar' onSlotchange={this._onTabsSlotChange}></slot>
+        </div>
+
+        <div class='tab-content'>
+          <slot onSlotchange={throttle(this._onContentSlotChange, 250)}></slot>
+        </div>
+      </Host>
+    );
+  }
+
+  public componentWillLoad(): void {
+    this.tabs = this._getTabs();
+    this.tabs.forEach((tab) => this._configure(tab));
+    this._initSelection();
+  }
+
+  public disconnectedCallback(): void {
+    this._observer.disconnect();
+  }
+
+  private _onContentSlotChange = (): void => {
+    const newTabs = this._getTabs()
+      .filter((tab) => !this.tabs.includes(tab));
+
+    // if a new tab/content is added to the tab group
+    if (newTabs.length) {
+      newTabs.forEach((tab) => this._configure(tab));
+      this.tabs = this.tabs.concat(newTabs);
+    }
+  };
+
+  private _onTabsSlotChange = (): void => {
+    const newTabs = this._getTabs();
+
+    // if a tab is removed from the tab group
+    if (newTabs.length < this.tabs.length) {
+      const removedTabs = this.tabs.filter((tab) => !newTabs.includes(tab));
+
+      removedTabs.forEach((removedTab) => {
+        removedTab.relatedContent?.remove();
+      });
+      this.tabs = newTabs;
+    }
+  };
+
+  private _getTabs(): InterfaceLyneTabGroupTab[] {
+    return (Array.from(this._element.children)
+      .filter((child) => (/^H\d$/u).test(child.tagName)) as InterfaceLyneTabGroupTab[]);
+  }
+
+  private _getEnabledTabs(): InterfaceLyneTabGroupTab[] {
+    return this.tabs.filter((t) => !t.hasAttribute('disabled'));
+  }
+
+  private _nextUId(): string {
+    return `lt${++this._lastUId}`;
+  }
+
+  private _ensureId(el): string {
+    el.id = el.id || this._nextUId();
+
+    return el.id;
+  }
+
+  private _initSelection(): void {
+    if (this.selectedIndex >= 0 && this.selectedIndex < this.tabs.length) {
+      if (this.tabs[this.selectedIndex].hasAttribute('disabled')) {
+        this._getEnabledTabs()[0]?.tabGroupActions.toggle();
```

Comment:
The .toggle() method name does not really show what would be toggled, maybe there is a better name?

Reply:
You're right, this method essentially selects a new tab, so I changed it to .select(). What do you think?